src/contracts/qpi.h

```diff
@@ -6285,7 +6285,7 @@ namespace QPI
 	#define PUBLIC_FUNCTION_WITH_LOCALS(function) \
 		public: \
 			enum { __is_function_##function = true }; \
-			static void function(const QPI::QpiContextFunctionCall& qpi, const CONTRACT_STATE_TYPE& state, function##_input& input, function##_output& output, function##_locals& locals) { ::__FunctionOrProcedureBeginEndGuard<(CONTRACT_INDEX << 22) | __LINE__> __prologueEpilogueCaller;
+			static void function(const QPI::QpiContextFunctionCall& qpi, CONTRACT_STATE_TYPE& state, function##_input& input, function##_output& output, function##_locals& locals) { ::__FunctionOrProcedureBeginEndGuard<(CONTRACT_INDEX << 22) | __LINE__> __prologueEpilogueCaller;

```

Comment:
Please undo this change in qpi.h. The file qpi.h cannot be changed by contract developers.

Reply:
I know that but if there is "const", the state values can't be used in UBLIC_FUNCTION. Please check again

Reply:
of course we can't change the value of state in PUBLIC_FUNCTION. but we need to get the value of state using PUBLIC_FUNCTION.

Reply:
could you reply for me if you checked my comment?

Reply:
> I know that but if there is "const", the state values can't be used in PUBLIC_FUNCTION. Please check again

FUNCTION == VIEW-ONLY
PROCEDURE == CAN MODIFY STATE

Reply:
currently FUNCTION != VIEW-ONLY, can't view the value of state in FUNCTION, please check again!

Reply:
I know that we can't modify the value of state in FUNCTION, I want to view the value of state in FUNCTION but using the state in FUNCTION occurs the syntax error. can't compile the project.

Reply:
> > I know that but if there is "const", the state values can't be used in PUBLIC_FUNCTION. Please check again
> 
> FUNCTION == VIEW-ONLY PROCEDURE == CAN MODIFY STATE

Please reply if you check